MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -1133,6 +1153,12 @@ namespace WPEFramework {
                     LOGERR("Maintenance is in Progress, Mode change not allowed");
                     result =true;
                 }
+#if defined(USE_IARMBUS) || defined(USE_IARM_BUS)
```

Comment:
this should go inside if if condition where the mode is getting updated. we need to send event only when it changes. Any caller can make use of getMaintenanceMode() if it doesn't want to persist the mode.

Reply:
Here i am restricting MaintenanceMode change means if the rfc is true then only "MaintenanceMode" change will work. Because in our discussion you have mention that maintenance mode change is not allowed during maintenance due to firmware download script. Now also we have firmware download via script and C method. And this is taken care based on the rfc value. That's why i have added rfc check.